src/main/java/org/matsim/contrib/gtfs/LineBrowser.java

```diff
@@ -119,40 +122,41 @@ private void tripStage(GTFSFeed feed, MyPattern pattern, Stage stage) {
 		return patternTreeItemTreeItem;
 	}
 
-	private class RootPatternTreeItem extends TreeItem {
+	private class RootPatternTreeItem extends TreeItem<RootPattern> {
 
 		public RootPatternTreeItem(GTFSFeed feed) {
 			super(new RootPattern());
-			Map<String, TreeItem> agencyTreeItems = new HashMap<>();
+			final Map<String, TreeItem<RootPattern>> agencyTreeItems = new HashMap<>();
 			for (Agency agency : feed.agency.values()) {
-				TreeItem agencyTreeItem = new TreeItem(new MyAgency(agency));
+				final TreeItem<RootPattern> agencyTreeItem = new TreeItem<>(new MyAgency(agency));
 				agencyTreeItems.put(agency.agency_id, agencyTreeItem);
 				getChildren().add(agencyTreeItem);
 			}
-			Map<String, TreeItem> routeTreeItems = new HashMap<>();
+			final Map<String, Pair<TreeItem<RootPattern>, MyRoute>> routeTreeItems = new HashMap<>();
 			for (Route route : feed.routes.values()) {
-				TreeItem routeTreeItem = new TreeItem(new MyRoute(route));
-				routeTreeItems.put(route.route_id, routeTreeItem);
+				final MyRoute myRoute = new MyRoute(route);
+				final TreeItem<RootPattern> routeTreeItem = new TreeItem<>(myRoute);
+				routeTreeItems.put(route.route_id, new Pair<>(routeTreeItem, myRoute));
 				agencyTreeItems.get(route.agency_id).getChildren().add(routeTreeItem);
 			}
 			for (Pattern pattern : feed.patterns.values()) {
-				for (String routeId : pattern.associatedRoutes) {
-					((MyRoute) routeTreeItems.get(routeId).getValue()).getPatterns().add(pattern);
+				for (String routeId : pattern.associatedTrips.stream().map(it -> feed.trips.get(it).route_id).collect(Collectors.toSet())) {
```

Comment:
This is the main line that needed changing from v1.1.0 to v5.0.7 of `gtfs-lib`. I looked at
https://github.com/conveyal/gtfs-lib/blob/1438918bda9506727317c8328287d1410b1b7071/src/main/java/com/conveyal/gtfs/model/Pattern.java#L43-L46
and replaced `associatedRoutes` with something similar.

The other changed lines in this file are introducing generics to the `TreeItem`s.